kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -67,115 +65,89 @@ public final class VirtualKafkaClusterReconciler implements
-                    .map(ResourcesUtil::toLocalRef)
-                    .collect(Collectors.toCollection(TreeSet::new));
-
-            TreeSet<ProxyRef> ingressProxyRefs = existingIngresses.stream().map(i -> i.getSpec().getProxyRef()).collect(Collectors.toCollection(TreeSet::new));
-            // if the ingress references a different proxy to the virtual cluster, we will not have loaded the proxy.
-            // TODO can we discriminate between unresolved this mismatch?
-            TreeSet<ProxyRef> mismatchedIngressProxyRef = ingressProxyRefs.stream().filter(i -> !existingProxyRefs.contains(i))
-                    .collect(Collectors.toCollection(TreeSet::new));
-
-            if (unresolvedServices.isEmpty()
-                    && unresolvedIngresses.isEmpty()
-                    && unresolvedFilters.isEmpty()
-                    && mismatchedIngressProxyRef.isEmpty()) {
-                uc = context
-                        .getSecondaryResource(ConfigMap.class, PROXY_CONFIG_MAP_EVENT_SOURCE_NAME)
-                        .flatMap(cm -> Optional.ofNullable(cm.getData()))
-                        .map(ProxyConfigData::new)
-                        .flatMap(data -> data.getStatusPatchForCluster(ResourcesUtil.name(cluster)))
-                        .map(patch -> {
-                            var rr = ResourceState.of(statusFactory.newTrueCondition(cluster, Condition.Type.ResolvedRefs));
-                            var acc = ResourceState.fromList(patch.getStatus().getConditions());
-                            return statusFactory.clusterStatusPatch(cluster, rr.replacementFor(acc));
-                        })
-                        .map(UpdateControl::patchStatus)
-                        .orElse(UpdateControl.noUpdate());
-
-            }
-            else {
-                Stream<String> serviceMsg = refsMessage("spec.targetKafkaServiceRef references ", cluster, unresolvedServices);
-                Stream<String> ingressMsg = refsMessage("spec.ingressRefs references ", cluster, unresolvedIngresses);
-                Stream<String> filterMsg = refsMessage("spec.filterRefs references ", cluster, unresolvedFilters);
-                Stream<String> ingressProxyMessage = refsMessage("a spec.ingressRef had an inconsistent or missing proxyRef ", cluster, mismatchedIngressProxyRef);
-                uc = UpdateControl
-                        .patchStatus(statusFactory.newFalseConditionStatusPatch(cluster, Condition.Type.ResolvedRefs, Condition.REASON_TRANSITIVE_REFS_NOT_FOUND,
-                                joiningMessages(serviceMsg, ingressMsg, filterMsg, ingressProxyMessage)));
-            }
+        ResolutionResult resolutionResult = resolver.deepResolveCluster(cluster, context);
+        ResolutionResult.ClusterResolutionResult clusterResolutionResult = resolutionResult.clusterResult(cluster)
+                .orElseThrow(() -> new IllegalStateException("no resolution result for cluster " + name(cluster)));
```

Comment:
Feels like a really funky API. `var result = deepResolve(cluster)` then `result.resultFor(the same thing as the previous line)`. Surely the result already know what cluster we are interested in?

Reply:
Needs a refactor I think, `ResolutionResult` contains all the resolved resources and then a `ClusterResolutionResult` that describes any problems with each cluster's dependencies. It was made for the KafkaProxy so doesn't quite fit this resolve cluster side.

Maybe we should shuffle the relevant dependencies down onto `ClusterResolutionResult` so that it carries the dependencies for each cluster, and or the description of dependencie issues. `deepResolveCluster` would return a ClusterResolutionResult.

Reply:
since we've already got enough going on here, I'd push that out to another PR

Reply:
this was fixed before merge